iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -24,15 +24,6 @@ public sealed class IotHubConnectionStringBuilder
         private const string SharedAccessSignaturePropertyName = "SharedAccessSignature";
         private const string GatewayHostNamePropertyName = "GatewayHostName";
 
-        // For some reason, the .NET SDK originally expected "X509Cert=true" in a connection string to inform the SDK that it would not
-        // include a shared access key, and to not error when parsing a connection string. However, the other SDK languages all followed
-        // the same key/value pair of "x509=true".
-        // So now we're adding support for it to work either way, so we stay compliant with the past but can improve documentation so all
-        // SDK languages refer to the same key/value pair naming.
-        private const string X509CertPropertyName = "X509Cert";
-
-        private const string CommonX509CertPropertyName = "x509";
```

Comment:
The Azure portal no longer provides connection strings for X509 authenticated clients. Removing this support.

Reply:
Azure IoT explorer does

Reply:
I know. I thought about that but since Azure portal doesn't, I took that as the best recommended official approach (since that is where everyone is going to start). Do you think this flow is worth preserving then? Or maybe, should we sync up with the rest of the SDKs and see how we collectively feel about it?

Reply:
Our constructors support only 2 flows today - either pass in a connection string, or pass in a IAUthenticationMethod that has either sas key/sas token information or the certificate information. There isn't a way for someone to supply both the certificates and a connection string. So, we'd have to go out of our way and modify our client initializers to support "x509=true" in connection string.

Reply:
The only reason for preserving it is for validating a connection string and throwing when it is invalid. If it doesn't have a SAK, it is invalid, unless it says, "oh but i'm using x509, trust me".

But we're not doing that, so I don't mind losing it.

I'm just saying it isn't gone from everywhere.